### DISCUSSION TITLE
Create migration file without applying it

### DISCUSSION THREAD

**Discussion by siggibk** [prisma/prisma#9691] [Q&A] [answered]
Not sure If I'm missing something from the documentation but is there no way of generating a migration file using a command like `prisma migrate dev` except not applying it to the database? Feels weird that it's applied generated and applied instantly without the chance of reviewing the generated migration file.

**Comment by TasinIshmam** [comment, marked as answer]
You can use the `--create-only` option to generate the migration file without applying it to the database. So the command will look like this:
```shell
npx prisma migrate dev --create-only
```
You can find more details in the [reference for Prisma Migrate](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-dev) in the docs.

**Reply by NicolasEspiau-stilll** [reply]
it emptied all my tables...

**Reply by aadilalkaype** [reply]
it also creates an empty migration file when there is no schema change. Any way to do "create the migration file for the schema changes, do not apply it. do not create a new file if no schema changes are there for which a migration file hasn't already been created"

**Reply by sidharthv96** [reply]
I have proposed to add a `--skip-empty` flag in https://github.com/prisma/prisma/pull/25180, which, when used with `--create-only`, will skip creating empty migration files.

**Reply by TanguyP** [reply]
> it emptied all my tables...

That's one scary story... but that didn't happen to me. `prisma migrate dev --create-only --name my-migration` only created an SQL file, then `prisma migrate dev` applied my migration on the (remote) DB, without touching any data or unexpected part of the schema. Tested on Prisma 7.8.0.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by marcospgp** [comment]
shouldn't not deploying automatically be the default? especially for production, one should inspect the migration carefully before applying
